Sources/Components/CollpsibleContentView/CollpsibleContentView.swift

```diff
@@ -0,0 +1,181 @@
+        addContentView(contentView)
+        if showExpanded {
+            expandContent()
+        }
+    }
+
+    // MARK: - Private methods
+
+    private func setup() {
+        clipsToBounds = true
+
+        headerView.addArrangedSubview(titleLabel)
+        headerView.addArrangedSubview(collapseIndicatorImageView)
+
+        innerContainerView.addSubview(headerView)
+        addSubview(innerContainerView)
+
+        NSLayoutConstraint.activate([
+            innerContainerView.topAnchor.constraint(equalTo: topAnchor),
+            innerContainerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            innerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            innerContainerView.bottomAnchor.constraint(greaterThanOrEqualTo: bottomAnchor),
+
+            headerView.topAnchor.constraint(equalTo: innerContainerView.layoutMarginsGuide.topAnchor),
+            headerView.leadingAnchor.constraint(equalTo: innerContainerView.layoutMarginsGuide.leadingAnchor),
+            headerView.trailingAnchor.constraint(equalTo: innerContainerView.layoutMarginsGuide.trailingAnchor),
+            compactHeightConstraint,
+        ])
+
+        headerView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(toggleContent)))
+    }
+
+    private func addContentView(_ contentView: UIView) {
+        innerContainerView.addSubview(contentView)
+
+        let contentTopConstraint = contentView.topAnchor.constraint(equalTo: headerView.bottomAnchor, constant: .mediumLargeSpacing)
+        NSLayoutConstraint.activate([
+            contentView.heightAnchor.constraint(greaterThanOrEqualToConstant: 10),
+            contentTopConstraint,
+            contentView.leadingAnchor.constraint(equalTo: innerContainerView.layoutMarginsGuide.leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: innerContainerView.layoutMarginsGuide.trailingAnchor),
+            contentView.bottomAnchor.constraint(equalTo: innerContainerView.layoutMarginsGuide.bottomAnchor),
+        ])
+        self.contentTopConstraint = contentTopConstraint
+    }
+
+    @objc private func toggleContent() {
+        guard expandsAutomatically else {
+            delegate?.collapsibleContentViewDidTapToggleContent(self)
+            return
+        }
+
+        let newImage = indicatorImage
+        UIView.transition(
+            with: collapseIndicatorImageView,
+            duration: 0.1,
+            options: [.transitionCrossDissolve],
+            animations: { self.collapseIndicatorImageView.image = newImage }
+        )
+
+        fullHeightConstraint.isActive.toggle()
+        UIView.animate(
+            withDuration: 0.3,
+            delay: 0,
+            options: .curveEaseInOut,
+            animations: { self.superview?.layoutIfNeeded() }
+        )
+    }
+
+    private func expandContent() {
+        fullHeightConstraint.isActive = true
+        collapseIndicatorImageView.image = UIImage(named: .arrowUp)
```

Comment:
Is it possible to use `indicatorImage` for this, and the one below, so we don't forget these if we change icons at some point? 😄